### PR TITLE
Issue #15: pspm_get_events.m bug with 1st row markers

### LINKED ISSUE #15: pspm_get_events misses markers that correspond to the first sample row
### Summary
`pspm_get_events` is not able to locate a marker if it occurs on the first sample in a given datafile. This frequently happens with viewpoint data where the marker on the first line indicates that data recording has begun.

### Steps to Reproduce
1. Let a viewpoint datapath be stored in `fp`. Make sure that `fp` contains only one marker on the first sample line and no others. This happens quite frequently with viewpoint data.
2.  Run
```
import{1}.type = 'marker';
options.overwrite = true;
pspm_import(fp, 'viewpoint', import, options);
```

### Expected Results
No error is produced and the data marker channels is imported without any errors/warnings.

### Actual Results
A warning from `pspm_get_events.m` line 75 is issued. `lo2hi` and `hi2lo` are empty whereas they both should contain `1`, the index of the marker.

### Technical Info
* PsPM revision ID: 2a6dda7
* MATLAB version: R2018b

### PR DESCRIPTION
**Pull request by irojkov-ph**
Fixes #15 :

Changes proposed in this pull request:
- The change is only a line (n°48) which is in a sense just avoiding the issue.
- It is detecting the first row's marker only for the `strcmpi(import.flank, 'ascending')` case otherwise it is not :(

### CODE REVIEW

**Comment by irojkov-ph** [comment]
I am still looking for a better method than that but for this I have to speak to you for about the issue itself. Because maybe I am not understanding it correctly.

**Comment by eozd** [comment]
Ok. I will check the implementation a bit myself, and we can discuss the issue on monday.

**Comment by eozd** [comment]
A few comments:

1. I think using `0` in the beginning and in the end is a good idea to catch markers in the first or in the last rows. However, the definitions of other variables that are affected by this change should also be fixed. (For example `finite`)
2. As we have discussed, `viewpoint` import currently returns these `+,=,+` lines, as well. This is in my opinion a bug; therefore, I will create a new issue about it and reference it here. We will need to fix that first, and only then merge this.
3. I checked other import types, but it seems like this is more of a viewpoint related issue. Still, ensuring that a possible marker in the first or in the last line is not missed is good practice.

**Comment by irojkov-ph** [comment]
Okay got it I will concentrate on #45 and look at that issue afterwards

**Comment by irojkov-ph** [comment]
Shall i merge this branch now ? We will anyway modify 'pspm_get_events` according to #31.

File: src/pspm_get_events.m [hunk 1]
```diff
@@ -45,7 +45,7 @@
     end
     
     % add more data in order to prevent deleting values with diff
-    data = [NaN; NaN; NaN; data; NaN; NaN; NaN;];
+    data = [0; 0; 0; data; 0; 0; 0;];
     % store information about finite and infinite in vector
     % used to reduce temp vector to relevant data
     finite = ~isnan(data);
```
Comment (eozd) [inline src/pspm_get_events.m]: Could you redefine `finite` according to the new definition of `data`?

**Comment by irojkov-ph** [comment]
This is one solution that I propose, it's not ideal but at least it will catch the first and last evt.

**Comment by irojkov-ph** [comment]
Yeah sorry:
- For data containing evt between `[2:end]` (i.e. no evt in `data(1)`) the program will replace `data` by `[NaN; NaN; NaN; data; 0;NaN;NaN]` (the 0 after data ensure you to catch the last evt as well if it is present otherwise it will not alter the execution of the code)

- For data having a data in the first position, the program will replace `data` by   `[NaN; head; head; data; 0;NaN;NaN]` with `head` being a random negative integer btw 0 and 1000 (I haven't set it to a precise number because i do not know what kind of evt one can have in data, if you ensure me that data is always >= 0 one could set head to -1). Then I've modified `finite` in order to avoid it taking head as a data point.

**Comment by eozd** [comment]
Hmm, I am bit confused about this. I think `data` is always a boolean vector, no? In that case, why doesn't the previous solution you implemented (i.e. `data = [0;0;0;data;0;0;0];`) work, and we need something like this?

**Comment by irojkov-ph** [comment]
I don't know if it is always a boolean. If yes then this solution could work: `data = [NaN; 0; 0; data; 0; 0; NaN;];`

The previous solution with `data = [0;0;0;data;0;0;0];` is not ideal because it will not catch the last evt (i.e. `data(end)`) when you use 'descending' parameter.

**Comment by eozd** [comment]
It seems like it has to be a boolean array to me. Otherwise, the previous code would essentially pick every single datapoint as a marker since all the neighbouring samples in a real-valued signal are almost surely unequal. Of course this corresponds to `'continuous'` option; when it is `'timestamps'`, we don't do anything anyways.

**Comment by irojkov-ph** [comment]
Okay then as said this solution is working well, should I merge ? 
(You can test if you want with the following lines and play a little bit with 'flank' parameter)
```
import.marker = 'continuous';
import.flank = 'ascending';
% import.flank = 'descending';
import.sr = 10^3;
import.data = zeros(10000,1);
idx_mrk__one = randi([1 4999],1,5);
idx_mrk__two = randi([5000 10000],1,5);
idx_mrk__three = 3001:1:3056;
import.data(idx_mrk__one) = 1;
import.data(idx_mrk__two) = 1;
import.data(idx_mrk__three) = 1;
import.data(1)=1;
import.data(end)=1;
[sts, rimport] = pspm_get_events(import);
```

**Pull request merged**